compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -663,8 +663,14 @@ pub(super) fn collect_return_position_impl_trait_in_trait_tys<'tcx>(
     // RPITs.
     let errors = ocx.select_all_or_error();
     if !errors.is_empty() {
-        let reported = infcx.err_ctxt().report_fulfillment_errors(errors);
-        return Err(reported);
+        if let Some(guar) =
+            try_report_async_mismatch(tcx, infcx, &errors, trait_m, impl_m, impl_sig)
```

Comment:
change `try_report_async_mismatch` to return `Result<(), ErrorGuaranteed>` and use `?` here.

Reply:
🧠